src/table/virtualized-table/hooks/use-scrollbar-width.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import useOnPropsChange from './use-on-props-change';
+
+const useScrollbarWidth = (ref: React.RefObject<HTMLDivElement>) => {
+  const [xScrollbarWidth, setXScrollbarWidth] = useState(0);
+  const [yScrollbarWidth, setYScrollbarWidth] = useState(0);
+  const { offsetWidth = 0, offsetHeight = 0, clientWidth = 0, clientHeight = 0 } = ref.current ?? {};
+  const widthDiff = offsetWidth - clientWidth;
+  const heightDiff = offsetHeight - clientHeight;
+
+  useEffect(() => {
+    if (ref.current) {
+      setYScrollbarWidth(ref.current.offsetWidth - ref.current.clientWidth);
+      setXScrollbarWidth(ref.current.offsetHeight - ref.current.clientHeight);
+    }
+  }, [ref]);
+
+  useOnPropsChange(() => {
```

Comment:
so the useEffect runs when the ref changes, I guess when the object size becomes bigger for some reason? And useOnPropsChange runs if the scrollbar appears?

Reply:
The `useEffect` is there to trigger the first check for a scrollbar. It's in a useEffect with `ref` as a dep because we want to trigger it first when the `Table` (where the ref is coming from) to have been rendered. `useOnPropsChange` would not be able to handle that case.

`useOnPropsChange` will then trigger on all other renders and update scrollbar width if needed.